sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -343,4 +354,144 @@ export class AssessmentSequentialProcessingService {
       options?.alternativeReferenceDate,
     );
   }
+
+  /**
+   * Get SFAS application part time awards data from either the SFAS application table or
+   * the SFAS part time applications table depending on the SFAS application query type.
+   * @param assessmentId Current assessment being processed.
+   * @param sfasApplicationType SFAS application query type (sfasApplication or sfasPTApplication).
+   * @returns SFAS application part time awards data.
+   */
+  async getProgramYearLegacyAwardsTotals(
+    assessmentId: number,
+    sfasApplicationType: string,
+  ): Promise<SFASApplicationAwards> {
+    const assessment = await this.studentAssessmentRepo.findOne({
+      select: {
+        id: true,
+        application: {
+          id: true,
+          applicationNumber: true,
+          student: {
+            id: true,
+            birthDate: true,
+            sinValidation: {
+              sin: true,
+            },
+            user: {
+              lastName: true,
+            },
+          },
+          programYear: {
+            id: true,
+            startDate: true,
+          },
+        },
+        assessmentDate: true,
+      },
+      relations: {
+        application: {
+          student: { sinValidation: true, user: true },
+          programYear: true,
+        },
+      },
+      where: {
+        id: assessmentId,
+      },
+    });
```

Comment:
This logic is almost the repetition of  line 155 -  178 in the existing method. 
![image](https://github.com/bcgov/SIMS/assets/54600590/1584baf4-2073-4d3b-894f-14d38baa016c)

 I would advice to re-use the same code by calling `getProgramYearLegacyAwardsTotals()` after line 178.